modules/openapi-generator/src/main/resources/tiny-cpp-client/model-header.mustache

```diff
@@ -0,0 +1,73 @@
+
+{{#models}}{{#model}}/*
+ * {{classname}}.h
+ *
+ * {{description}}
+ */
+
+#ifndef _{{classname}}_H_
+#define _{{classname}}_H_
+
+{{/model}}{{/models}}
+#include <string>
+#include "bourne/json.hpp"
+#include "Helpers.h"
+{{#imports}}
+{{{import}}}
+{{/imports}}
+
+namespace Tiny {
+{{#models}}{{#model}}
+
+/*! \brief {{{description}}}
+ *
+ *  \ingroup Models
+ *
+ */
+
+class {{classname}}{
+public:
+
+    /*! \brief Constructor.
+	 */
+    {{classname}}();
+    {{classname}}(std::string jsonString);
+
+
+    /*! \brief Destructor.
+	 */
+    virtual ~{{classname}}();
+
+
+    /*! \brief Retrieve a bourne JSON representation of this class.
+	 */
+    bourne::json toJson();
+
+
+    /*! \brief Fills in members of this class from bourne JSON object representing it.
+	 */
+    void fromJson(std::string jsonObj);
+
+    {{#vars}}
+	/*! \brief Get {{{description}}}
+	 */
+	{{dataType}}{{#isContainer}}{{#isMap}}<std::string, std::string>{{/isMap}}{{^isMap}}<{{#items}}{{dataType}}{{/items}}>{{/isMap}}{{/isContainer}} {{getter}}();
+
+	/*! \brief Set {{{description}}}
+	 */
+	void {{setter}}({{dataType}} {{#isContainer}}{{#isMap}}<std::string, std::string>{{/isMap}}{{^isMap}}<{{#items}}{{dataType}}{{/items}}>{{/isMap}}{{/isContainer}} {{name}});
+	{{/vars}}
+
+
+    private:
+    {{#vars}}
+	{{dataType}} {{#isContainer}}{{#isMap}}<std::string, std::string>{{/isMap}}{{^isMap}}<{{#items}}{{dataType}}{{/items}}>{{/isMap}}{{/isContainer}}{{name}};
+	{{/vars}}
+    void __init();
```

Comment:
Using double underscore is undefined behavior, as specified by C++ standard. Problem is that standard library implementation can use something like `#define __init whatever` - it's what C++ standard allows them to do.

I'd change to `init()` for simplicity. If you really want an underscore for private members, then I'd put it at the end `init_()`

https://stackoverflow.com/questions/228783/what-are-the-rules-about-using-an-underscore-in-a-c-identifier

Reply:
We've removed init completely :)